pubsub/pubsub-aws/src/main/java/com/salesforce/multicloudj/pubsub/aws/AwsTopic.java

```diff
@@ -325,44 +565,96 @@ public Builder builder() {
     }
     
     public static class Builder extends AbstractTopic.Builder<AwsTopic> {
+        private ServiceType serviceType; 
         private SqsClient sqsClient;
-        private String topicUrl;
+        private SnsClient snsClient;
+        private String topicUrl;  // For SQS
+        private String topicArn;   // For SNS
+        private TopicOptions topicOptions;
         
         public Builder() {
             this.providerId = AwsConstants.PROVIDER_ID;
         }
         
+        public Builder withServiceType(ServiceType serviceType) {
```

Comment:
this doesn't look ideal, the end user will have to use SQS and SNS ? these are very aws specific. 
we should have some cloud agnostic terms